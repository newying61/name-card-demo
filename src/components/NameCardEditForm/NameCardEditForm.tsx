import React from 'react';
import Form from 'react-bootstrap/Form';
import { StyledButton, ButtonsWrapper } from '../Styled/Buttons';
import { FieldNameMap } from '../../constants';
import FormField from '../FormField';

const NameCardEditForm: React.FC<any> = ({
    values,
    touched,
    errors,
    handleChange,
    handleBlur,
    handleSubmit,
    isSubmitting,
}) => {
    return (
      <Form noValidate onSubmit={handleSubmit}>
        <FormField
          name="firstName"
          label={FieldNameMap.firstName}
          inputType="text"
          placeHolder=""
          value={values.firstName}
          handleChange={handleChange}
          errors={errors}
        />
        <FormField
          name="lastName"
          label={FieldNameMap.lastName}
          inputType="text"
          placeHolder=""
          value={values.lastName}
          handleChange={handleChange}
          errors={errors}
        />
        <FormField
          name="dateOfBirth"
          label={FieldNameMap.dateOfBirth}
          inputType="date"
          placeHolder=""
          value={values.dateOfBirth}
          handleChange={handleChange}
          errors={errors}
        />
        <ButtonsWrapper>
            <StyledButton type="submit">Submit</StyledButton>
            <StyledButton>Cancel</StyledButton>
        </ButtonsWrapper>
    </Form>
    );
};

export default NameCardEditForm;
