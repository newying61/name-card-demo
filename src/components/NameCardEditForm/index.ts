import NameCardEditForm from './NameCardEditForm';

export default NameCardEditForm;
