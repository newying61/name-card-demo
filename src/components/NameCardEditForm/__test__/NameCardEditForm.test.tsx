import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NameCardEditForm from '../';

afterEach(cleanup);

describe('AppsConfig', () => {
    it('should render without crash', () => {
      const { container } = render(
        <NameCardEditForm
          values={{}}
          touched={{}}
          errors={{}}
          handleChange={jest.fn()}
          handleBlur={jest.fn}
          handleSubmit={jest.fn}
          isSubmitting={false}
       />
      );

      expect(container.firstChild).toMatchSnapshot();
  });
});

// TODO: Add more tests for the form
