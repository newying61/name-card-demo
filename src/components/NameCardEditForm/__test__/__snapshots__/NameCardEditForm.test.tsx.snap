// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppsConfig should render without crash 1`] = `
<form
  class=""
  novalidate=""
>
  <div
    class="form-group"
  >
    <label
      class="form-label"
      for="firstName-field"
    >
      First name
    </label>
    <input
      class="sc-htpNat eODREG form-control"
      id="firstName-field"
      name="firstName"
      placeholder=""
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      class="form-label"
      for="lastName-field"
    >
      Last name
    </label>
    <input
      class="sc-htpNat eODREG form-control"
      id="lastName-field"
      name="lastName"
      placeholder=""
      type="text"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      class="form-label"
      for="dateOfBirth-field"
    >
      Date of birrth
    </label>
    <input
      class="sc-htpNat eODREG form-control"
      id="dateOfBirth-field"
      name="dateOfBirth"
      placeholder=""
      type="date"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
  <div
    class="sc-bwzfXH fpBhud"
  >
    <button
      class="sc-bdVaJa jElpOZ btn btn-primary"
      type="submit"
    >
      Submit
    </button>
    <button
      class="sc-bdVaJa jElpOZ btn btn-primary"
      type="button"
    >
      Cancel
    </button>
  </div>
</form>
`;
